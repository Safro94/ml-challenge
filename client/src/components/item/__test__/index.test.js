import React from 'react';

import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';

import formatCurrency from 'utils/currency';

import Item from '../';

jest.mock('utils/currency', () => jest.fn(() => '300,50'));

describe('Item', () => {
	const onClick = jest.fn();
	const item = {
		picture: 'a.jpg',
		title: 'title',
		price: {
			amount: 300,
			decimals: 50,
		},
		condition: 'new',
		address: 'address',
	};

	it('should render without free shipping', () => {
		//Act
		const { container } = render(<Item item={item} />);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should render with free shipping', () => {
		//Arrange
		item.free_shipping = true;

		//Act
		const { container } = render(<Item item={item} />);

		//Assert
		expect(container).toMatchSnapshot();
	});

	it('should call onclick', () => {
		//Act
		render(<Item item={item} onClick={onClick} />);
		userEvent.click(screen.getByTestId('item'));

		//Assert
		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalledWith(item.id);
	});

	it('should call formatCurrency', () => {
		//Act
		render(<Item item={item} />);

		//Assert
		expect(formatCurrency).toHaveBeenCalledTimes(1);
		expect(formatCurrency).toHaveBeenCalledWith({
			amount: item.price.amount,
			decimals: item.price.decimals,
		});
	});
});
