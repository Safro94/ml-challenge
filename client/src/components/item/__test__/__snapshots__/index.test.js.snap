// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render with free shipping 1`] = `
<div>
  <div
    class="container"
    data-testid="item"
  >
    <div>
      <img
        alt="title"
        class="itemImage"
        src="a.jpg"
      />
    </div>
    <div
      class="information"
    >
      <div>
        <h2
          class="price"
        >
          <span
            class="currencyCode"
          >
            $
          </span>
          300,50
          <img
            alt="Free shipping"
            class="shippingImage"
            src="/assets/ic_shipping@2x.png.png.png"
          />
        </h2>
        <h3>
          title
        </h3>
        <h3>
          new
        </h3>
      </div>
      <div>
        <span
          class="address"
        >
          address
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Item should render without free shipping 1`] = `
<div>
  <div
    class="container"
    data-testid="item"
  >
    <div>
      <img
        alt="title"
        class="itemImage"
        src="a.jpg"
      />
    </div>
    <div
      class="information"
    >
      <div>
        <h2
          class="price"
        >
          <span
            class="currencyCode"
          >
            $
          </span>
          300,50
        </h2>
        <h3>
          title
        </h3>
        <h3>
          new
        </h3>
      </div>
      <div>
        <span
          class="address"
        >
          address
        </span>
      </div>
    </div>
  </div>
</div>
`;
